CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java

```diff
@@ -126,19 +126,30 @@ public NumberInputPane(final PluginParameter<?> parameter) {
         // Just typing doesn't fire value property change events, and doesn't allow us to change the style
         // when the string doesn't validate.
         field.getEditor().textProperty().addListener((final ObservableValue<? extends String> ov, final String oldValue, final String newValue) -> {
-            final String error = parameter.validateString(field.getValueFactory().getValue().toString());
-            if (error != null) {
-                tooltip.setText(error);
-                field.setTooltip(tooltip);
-                field.setId(INVALID_ID);
+            final int dotPos = newValue.indexOf(".");
+            final String intPart = dotPos > -1 ? newValue.substring(0, dotPos) : newValue;
+            final String decPart = dotPos > -1 ? newValue.substring(dotPos + 1) : "";
+            final boolean isIntVal = parameter.getType().getId().equals(IntegerParameterType.ID);
+            // Integers: Max 9 digits.  Floats: Max 8 digits before the decimal, and 2 digits after.
+            if (intPart.matches("[\\-]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?" + (isIntVal ? "[0-9]?" : "")) 
```

Comment:
could the series of [0-9]?... be reduced to [0-9]{0, 8}?
I could also be wrong but with that regex statement, technically "-" on its own will match (so you might need to specify that if there is a "-" then there needs to be something after it)